micrometer-commons/src/main/java/io/micrometer/common/annotation/AnnotationHandler.java

```diff
@@ -89,7 +89,7 @@ public AnnotationHandler(BiConsumer<KeyValue, T> keyValueConsumer,
     public void addAnnotatedParameters(T objectToModify, ProceedingJoinPoint pjp) {
         try {
             Method method = ((MethodSignature) pjp.getSignature()).getMethod();
-            method = pjp.getTarget().getClass().getMethod(method.getName(), method.getParameterTypes());
+            method = pjp.getTarget().getClass().getDeclaredMethod(method.getName(), method.getParameterTypes());
```

Comment:
No, I don't think that was intended. Could you open an issue for it? I guess we're missing a test case to cover that scenario since we didn't notice it.

Reply:
done: https://github.com/micrometer-metrics/micrometer/issues/4983

Reply:
@bpernetupgrade Ooops, no it wasn't intended and as @shakuzen says the lack of test coverage is at fault here.

Good found though, I'll see if I can provide a fix for it.

Reply:
@bpernetupgrade @shakuzen 

Please take a look at #4985